cpp/demo/interpolation/main.cpp

```diff
@@ -0,0 +1,80 @@
+#include <basix/e-lagrange.h>
+#include <cmath>
+#include <dolfinx/common/timing.h>
+#include <dolfinx/fem/utils.h>
+#include <dolfinx/io/ADIOS2Writers.h>
+#include <dolfinx/mesh/generation.h>
+#include <memory>
+
+using namespace dolfinx;
+
+using T = double;
+
+void interpolation_different_meshes()
+{
+  const std::array<std::size_t, 3> subdivisions = {50, 50, 50};
+
+  auto meshL = std::make_shared<mesh::Mesh>(mesh::create_box(
```

Comment:
We could use `GhostMode::none` here as we do not do any interior integrals in this demo